ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -65,11 +67,9 @@ let join_internal ~__context ~self =
       let pIF = Db.Cluster_host.get_PIF ~__context ~self in
       let pifrec = Db.PIF.get_record ~__context ~self:pIF in
       fix_pif_prerequisites ~__context (pIF,pifrec);
-
-      let host = Db.Cluster_host.get_host ~__context ~self in
-      assert_cluster_host_can_be_created ~__context ~host;
-      assert_pif_attached_to ~host ~pIF ~__context;
-      assert_pif_prerequisites (pIF,pifrec);
+      (* Re-acquire PIF record after it has updated *)
```

Comment:
What kind of delay do we need here in order to refresh the record?

Reply:
Or should we just use `pif_of_host` here again?

Reply:
No delay is needed, just fetch it again.

Reply:
Would be good if fix_pif_prerequisites fetched the record itself, to avoid accidentally reusing a stale record in the caller.

Reply:
I've updated all the PIF helpers to fetch the record themselves now.